### ISSUE TITLE
[Bug]: Some translation missing

### ISSUE THREAD

**Issue by TUCAOEVER** [nextcloud/server#44950]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

![image](https://github.com/nextcloud/server/assets/43756134/627e7b41-c065-469f-be90-0eab8112696b)
![image](https://github.com/nextcloud/server/assets/43756134/5acf2bb7-79a5-4689-97ed-fab3dd332cf1)

### Steps to reproduce

1.Viewing the page of file of sharing or editing time related
2.Viewing the page of file version

### Expected behavior

Message in the screenshots should be translable

### Installation method

Community Docker image

### Nextcloud Server version

28

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.0

### Web server

Nginx

### Database engine version

MySQL

### Is this bug present after an update or on a fresh install?

Updated from a MINOR version (ex. 22.1 to 22.2)

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```shell
{
    "system": {
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "*.*.*.*",
            "cloud.promc.cn"
        ],
        "dbtype": "mysql",
        "version": "28.0.4.1",
        "overwriteprotocol": "https",
        "overwrite.cli.url": "https:\/\/cloud.promc.cn",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpmode": "smtp",
        "mail_sendmailmode": "smtp",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpauth": 1,
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "465",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpsecure": "ssl",
        "allow_local_remote_servers": true,
        "theme": "",
        "loglevel": 0,
        "memories.exiftool": "\/config\/www\/nextcloud\/apps\/memories\/bin-ext\/exiftool-amd64-musl",
        "memories.vod.path": "\/config\/www\/nextcloud\/apps\/memories\/bin-ext\/go-vod-amd64",
        "memories.vod.ffmpeg": "\/usr\/bin\/ffmpeg",
        "memories.vod.ffprobe": "\/usr\/bin\/ffprobe",
        "maintenance_window_start": 8,
        "upgrade.disable-web": false,
        "app_install_overwrite": [
            "memories",
            "richdocuments",
            "files_rightclick"
        ],
        "maintenance": false,
        "memcache.local": "\\OC\\Memcache\\APCu",
        "memcache.distributed": "\\OC\\Memcache\\Redis",
        "filelocking.enabled": "true",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "timeout": 0,
            "dbindex": 0,
            "port": 6379,
            "password": "***REMOVED SENSITIVE VALUE***"
        },
        "memories.db.triggers.fcu": true,
        "default_language": "zh_CN",
        "default_locale": "zh-cn",
        "default_phone_region": "CN"
    }
}
```

### List of activated Apps

```shell
Enabled:
  - activity: 2.20.0
  - calendar: 4.6.7
  - circles: 28.0.0
  - cloud_federation_api: 1.11.0
  - comments: 1.18.0
  - contactsinteraction: 1.9.0
  - dashboard: 7.8.0
  - dav: 1.29.1
  - drawio: 3.0.2
  - federatedfilesharing: 1.18.0
  - federation: 1.18.0
  - files: 2.0.0
  - files_pdfviewer: 2.9.0
  - files_reminders: 1.1.0
  - files_sharing: 1.20.0
  - files_trashbin: 1.18.0
  - files_versions: 1.21.0
  - firstrunwizard: 2.17.0
  - logreader: 2.13.0
  - lookup_server_connector: 1.16.0
  - memories: 7.2.0
  - nextcloud_announcements: 1.17.0
  - notifications: 2.16.0
  - oauth2: 1.16.3
  - password_policy: 1.18.0
  - photos: 2.4.0
  - previewgenerator: 5.5.0
  - privacy: 1.12.0
  - provisioning_api: 1.18.0
  - recommendations: 2.0.0
  - related_resources: 1.3.0
  - richdocuments: 8.3.4
  - serverinfo: 1.18.0
  - settings: 1.10.1
  - sharebymail: 1.18.0
  - support: 1.11.1
  - survey_client: 1.16.0
  - systemtags: 1.18.0
  - text: 3.9.1
  - theming: 2.3.0
  - twofactor_backupcodes: 1.17.0
  - updatenotification: 1.18.0
  - user_status: 1.8.1
  - viewer: 2.2.0
  - workflowengine: 2.10.0
Disabled:
  - admin_audit: 1.18.0
  - bruteforcesettings: 2.8.0
  - encryption: 2.16.0
  - files_external: 1.20.0
  - files_rightclick: 0.15.1 (installed 0.15.1)
  - music: 1.11.0 (installed 1.11.0)
  - notes: 4.9.4 (installed 4.9.4)
  - suspicious_login: 6.0.0
  - twofactor_totp: 10.0.0-beta.2
  - user_ldap: 1.19.0
  - weather_status: 1.8.0 (installed 1.8.0)
```

**Comment by TUCAOEVER** [comment]
I found any locale without suffix could show correct date, ex. 'vi' 'fr' is alright, but 'zh_CN' not.

**Comment by susnux** [comment]
Files list is not using moment, but native browser localization. Note that it uses the locale not the language for date formatting.

I tested this and it correctly works on 29 and current master branch, even with suffixes like `zn_Hans`.

This might still be an issue for apps using legacy moment JS, for this the issue is tracked here:
https://github.com/nextcloud-libraries/nextcloud-moment/issues/889

**Comment by TUCAOEVER** [comment]
> Files list is not using moment, but native browser localization. Note that it uses the locale not the language for date formatting.
> 
> I tested this and it correctly works on 29 and current master branch, even with suffixes like `zn_Hans`.
> 
> This might still be an issue for apps using legacy moment JS, for this the issue is tracked here: [nextcloud-libraries/nextcloud-moment#889](https://github.com/nextcloud-libraries/nextcloud-moment/issues/889)

Even i got updated to 29.0.2, this issue still exists.
This dose not related to file listing but file version list.

![image](https://github.com/nextcloud/server/assets/43756134/14fb15a8-bb7a-4a89-9981-cd26dcc14dc6)

**Comment by TUCAOEVER** [comment]
> Files list is not using moment, but native browser localization. Note that it uses the locale not the language for date formatting.
> 
> I tested this and it correctly works on 29 and current master branch, even with suffixes like `zn_Hans`.
> 
> This might still be an issue for apps using legacy moment JS, for this the issue is tracked here: [nextcloud-libraries/nextcloud-moment#889](https://github.com/nextcloud-libraries/nextcloud-moment/issues/889)

Besides, anything but "a few seconds ago" phase in file list was not translated

**Comment by susnux** [comment]
@TUCAOEVER the issue with the missing translation for versions will be fixed with 29.0.4

**Comment by TUCAOEVER** [comment]
> @TUCAOEVER the issue with the missing translation for versions will be fixed with 29.0.4

Forgot to say, the modification date of the file sharing page has also not been translated
![image](https://github.com/nextcloud/server/assets/43756134/2888ad42-262c-4e84-8733-c05269ba7fcf)
![image](https://github.com/nextcloud/server/assets/43756134/5893efb8-4df1-41ad-9088-414774c0a378)
